internal/karma/karma.conf.js

```diff
@@ -3,7 +3,57 @@
 try
 {
   const fs = require('fs');
+  const path = require('path');
   const tmp = require('tmp');
+  const child_process = require('child_process');
+  const isWindows = /^win/i.test(process.platform);
+
+  // Helper function to find a particular namedFile
+  // within the webTestMetadata webTestFiles
+  function findNamedFile(webTestMetadata, key) {
+    let result;
+    webTestMetadata['webTestFiles'].forEach(entry => {
+      const webTestNamedFiles = entry['namedFiles'];
+      if (webTestNamedFiles && webTestNamedFiles[key]) {
+        result = webTestNamedFiles[key];
+      }
+    });
+    return result;
+  }
+
+  // Helper function to extract a browser archive
+  // and return the path to extract executable
+  function extractWebArchive(extractExe, archiveFile, executablePath) {
+    try {
+      // Paths are relative to the 'external' folder within runfiles
+      extractExe = extractExe ? path.join('external', extractExe) : extractExe;
+      archiveFile = path.join('external', archiveFile)
+      const extractedExecutablePath = path.join(process.cwd(), executablePath);
+      if (isWindows) {
+        // We can't run the EXTRACT_EXE shell script on Windows
+        const AdmZip = require('adm-zip');
+        const zip = new AdmZip(archiveFile);
+        // Extract to cwd
+        zip.extractAllTo('.', true);
+        // We need to chmod the executable to executable since adm-zip doesn't handle that
+        fs.chmodSync(extractedExecutablePath, 755);
+      } else {
+        // If we're on OSX or Linux then we can use the provided rules_webtesting
```

Comment:
could we do it with adm-zip always? why have two different modes if one of them is broken on windows

Reply:
that was my original plan. however on Linux the archive is a tar.xz which adm-zip doesn't handle